src/main/java/com/karankumar/booksapi/service/AuthorService.java

```diff
@@ -42,4 +51,17 @@ public Author save(@NonNull Author author) {
     public List<Author> findAll() {
         return authorRepository.findAllAuthors();
     }
+
+    public void deleteAuthor(@NonNull Author author) {
+        Set<Book> booksWithOneAuthor = author.getBooks()
```

Comment:
Now I think that maybe should I write query instead?

Reply:
Yes, that would be good. Thanks

Reply:
done